mosaic-runtime/src/commonMain/kotlin/com/jakewharton/mosaic/canvas.kt

```diff
@@ -26,16 +28,68 @@ internal interface TextCanvas {
 private val blankPixel = TextPixel(' ')
 
 internal class TextSurface(
-	override val width: Int,
-	override val height: Int,
 	private val ansiLevel: AnsiLevel,
 ) : TextCanvas {
+	override var width: Int = 0
+		private set
+	override var height: Int = 0
+		private set
+
 	override var translationX = 0
 	override var translationY = 0
 
-	private val rows = Array(height) { Array(width) { TextPixel(' ') } }
+	private var fullWidth: Int = width
+	private var fullHeight: Int = height
+
+	private val rows = ArrayList(fullHeight) { ArrayList(fullWidth) { TextPixel(' ') } }
```

Comment:
I don't see a big motivation from switching from `Array` here. Is it because of row resizing? We can accomplish that with a helper https://pl.kotl.in/c_ED2lc0h. I'll file a Kotlin issue to provide such an overload.

Reply:
https://youtrack.jetbrains.com/issue/KT-70984

Reply:
Yes, I changed it only to change the size out of the box.